src/foundation/components/Label/index.tsx

```diff
@@ -0,0 +1,73 @@
+import { cx } from '@linaria/atomic';
+import React, { ElementType, HTMLAttributes } from 'react';
+
+import { labelStyles } from './css';
+import { textColors, themedColorVars } from '../../colors/css';
+import { useLocalProps } from '../../hooks/useLocalProps';
+import { SBUFoundationProps } from '../../types';
+
+type AsProp<C extends ElementType> = {
+  as?: C;
+};
+type PropsToOmit<C extends ElementType, P> = keyof (P & AsProp<C>);
+type PolymorphicComponentProps<C extends ElementType, Props = NonNullable<unknown>> = Props &
+  AsProp<C> &
+  Omit<React.ComponentPropsWithoutRef<C>, PropsToOmit<C, Props>>;
+
+type HTMLProps<C extends ElementType> = PolymorphicComponentProps<C, HTMLAttributes<HTMLElement>>;
```

Comment:
here and above is a bit complex to understand. Let me ask you later.

Reply:
This is simply for accurate type inference. To fully understand it, you need to study the inference in TypeScript further. If you do not wish to deep dive into it, you can just understand it as a part for type inference.